traitsui/tests/ui_editors/test_data_frame_editor.py

```diff
@@ -0,0 +1,287 @@
+#  Copyright (c) 2015, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in enthought/LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+
+import nose
+from numpy.testing import assert_array_equal
+
+try:
+    from pandas import DataFrame
+except ImportError as exc:
+    print "Can't import Pandas: skipping"
+    raise nose.SkipTest
+
+from traits.api import HasTraits, Instance
+
+from traitsui.item import Item
+from traitsui.ui_editors.data_frame_editor import (
+    DataFrameEditor, DataFrameAdapter)
+from traitsui.view import View
+
+from traitsui.tests._tools import *
+
+
+class DataFrameViewer(HasTraits):
+
+    data = Instance('pandas.core.frame.DataFrame')
+
+    view = View(
+        Item('data', editor=DataFrameEditor())
+    )
+
+def sample_data():
+    data = [[ 0,  1,  2],
+            [ 3,  4,  5],
+            [ 6,  7,  8],
+            [ 9, 10, 11]]
+    df = DataFrame(data, index=['one', 'two', 'three', 'four'],
+                   columns=['X', 'Y', 'Z'])
+    viewer = DataFrameViewer(data=df)
+    return viewer
+
+def sample_data_numerical_index():
+    data = [[ 0,  1,  2],
+            [ 3,  4,  5],
+            [ 6,  7,  8],
+            [ 9, 10, 11]]
+    df = DataFrame(data, index=range(1,5),
+                   columns=['X', 'Y', 'Z'])
+    viewer = DataFrameViewer(data=df)
+    return viewer
+
+
+def test_adapter_get_item():
```

Comment:
We should add a unit test like this one for the case where the dataframe has no rows as it should be supported. It could be the initial state of a model that this editor is viewing.